gossipsub.go

```diff
@@ -176,29 +208,93 @@ func NewGossipSub(ctx context.Context, h host.Host, opts ...Option) (*PubSub, er
 		peerhave: make(map[peer.ID]int),
 		iasked:   make(map[peer.ID]int),
 		outbound: make(map[peer.ID]bool),
-		connect:  make(chan connectInfo, GossipSubMaxPendingConnections),
-		mcache:   NewMessageCache(GossipSubHistoryGossip, GossipSubHistoryLength),
+		connect:  make(chan connectInfo, params.GossipSubMaxPendingConnections),
+		mcache:   NewMessageCache(params.GossipSubHistoryGossip, params.GossipSubHistoryLength),
 
 		// these are configured per router to allow variation in tests
-		D:      GossipSubD,
-		Dlo:    GossipSubDlo,
-		Dhi:    GossipSubDhi,
-		Dscore: GossipSubDscore,
-		Dout:   GossipSubDout,
-		Dlazy:  GossipSubDlazy,
+		D:      params.GossipSubD,
+		Dlo:    params.GossipSubDlo,
+		Dhi:    params.GossipSubDhi,
+		Dscore: params.GossipSubDscore,
+		Dout:   params.GossipSubDout,
+		Dlazy:  params.GossipSubDlazy,
```

Comment:
let's just use the params struct directly, this is getting messy.

Reply:
Also, let's make the params object a plain struct, not a pointer so that there is no extra indirection.